### DISCUSSION TITLE
will json parse works for JSON schema inheritance

### DISCUSSION THREAD

**Discussion by praveensoma** [nlohmann/json#2246] [Q&A] [answered]
for example i have base.json with one std::string in it.

if i include it in derive.json and convert to c++ using quicktype.

what's happening is

```cpp
 class Id {
        public:
        Id() = default;
        virtual ~Id() = default;

        private:
        std::string _id;

        public:
        const std::string & get_id() const { return _id; }
        std::string & get_mutable_id() { return _id; }
        void set_id(const std::string & value) {  this->_id = value; }
    };


class Derived{
        public:
        Derived() = default;
        virtual ~Derived() = default;

        private:
        Id m_id;

        public:
        const Id & get_id() const { return m_id; }
        Id & get_mutable_id() { return m_id; }
        void set_id(const Id & value) { this->m_id = value; }

       
    };
```

what is the correct JSON object format for Derived class

**Comment by nlohmann** [comment, marked as answer]
I am not sure what you want to achieve.

**Reply by praveensoma** [reply]
namespace ns {
    // a simple struct to model a person
    struct person {
        std::string name;
        std::string address;
        int age;
    };
}
ns::person p {"Ned Flanders", "744 Evergreen Terrace", 60};
  
j = p worked here
in the above example if name is in another class name and if it becomes like below

    struct Name{
        std::string name;
     
    };

   struct person {
        Name na;
        std::string address;
        int age;
    };


will the same json::parse works here , how to work in those cases, can you please suggest

**Reply by nlohmann** [reply]
What did you try? Did it work? If not, what is the error message?

**Reply by praveensoma** [reply]
terminate called after throwing an instance of 'nlohmann::detail::type_error'
[json.exception.type_error.304] cannot use at() with string

**Reply by nlohmann** [reply]
And what is the code that you executed? Can you provide a small working example?

**Reply by nlohmann** [reply]
Assuming you want to extend the `person` struct with your own `Name` member, then you have to do two things:

1. Implement similar `to_json`/`from_json` functions for `Name`.
2. Extend the `to_json`/`from_json` functions for `person` to cope with the `na` member.

Both can be simplified with the macros `NLOHMANN_DEFINE_TYPE_INTRUSIVE` and `NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE` that we added recently, see https://github.com/nlohmann/json#simplify-your-life-with-macros.

**Reply by praveensoma** [reply]
sorry for the trouble, mistake was in our json schema, we converted it to c++, we fixed it.

Many Thanks for your time and reply
